### ISSUE TITLE
BEM

### ISSUE THREAD

**Issue by Hahahayabanan** [Liquense/FSD-task-2#5]
Чаще всего, в бэм используется плоская структура, так каждый блок описывается одним файлом js, scss, pug соответственно. Сейчас у тебя разбросаны модификаторы по разным папкам, что не очень удобно. Собрать стили для блока, его элементов, модификаторов в одном месте, будет лучше, мне кажется

**Comment by Liquense** [comment]
Не очень понял: получается, что все файлы, относящиеся к блоку, должны лежать условно в корневой папке этого блока?
Что-то типа такого:
![image](https://user-images.githubusercontent.com/20877460/85130230-bb768c80-b25e-11ea-95a5-cd5fa3efe7da.png)

Просто казалось, что древовидная структура файлов нагляднее воспринимается

**Comment by Hahahayabanan** [comment]
> Не очень понял: получается, что все файлы, относящиеся к блоку, должны лежать условно в корневой папке этого блока?
> Что-то типа такого:
> ![image](https://user-images.githubusercontent.com/20877460/85130230-bb768c80-b25e-11ea-95a5-cd5fa3efe7da.png)
> 
> Просто казалось, что древовидная структура файлов нагляднее воспринимается

Нет, лучше в всё одном файле, так сразу видно все стили, не нужно бегать из папки в папку и искать что-либо

**Comment by Hahahayabanan** [comment]
Неверная последовательность элемента и модификатора

![image](https://user-images.githubusercontent.com/43137324/86715890-b886e480-c031-11ea-9d40-00c1073f469f.png)

**Comment by Hahahayabanan** [comment]
Это блок? Что это и зачем здесь?

![image](https://user-images.githubusercontent.com/43137324/87968639-c9bffe80-cad1-11ea-8056-a8da641e8b9b.png)

**Comment by Liquense** [comment]
это файл с шаблоном, который создаётся во время инициализации плагина из JS-модуля, чтобы держать в JS-коде по минимуму html
конкретно в пончике наверное можно унести это всё в миксин, но в datepicker-е кнопки создаются после инициализации плагина календаря (datepicker-block_button-template.pug для них)

**Comment by Hahahayabanan** [comment]
> это файл с шаблоном, который создаётся во время инициализации экземпляра класса блока, чтобы держать в `JS`-коде по минимуму `html`-а
> 
> конкретно в пончике наверное можно унести это всё в миксин, но в `datepicker` кнопки создаются после инициализации плагина календаря (`datepicker-block_button-template.pug` для них, это заодно даёт не обновлять их руками, если что-то изменится в кнопках типа текст)

Всё же не должно быть несколько pug файлов для одного блока. Тут ты можешь, как минимум, стилизовать стандартные блоки, либо можешь добавить их динамически, оперируя методами js класса кнопки.

**Comment by Hahahayabanan** [comment]
Элементы вне блока

![image](https://user-images.githubusercontent.com/43137324/89116443-2c87a180-d4a5-11ea-97ca-8cbbfdadc01c.png)

**Comment by Hahahayabanan** [comment]
Используются элементы из другого блока

![image](https://user-images.githubusercontent.com/43137324/89116507-ca7b6c00-d4a5-11ea-8ce9-8cc27801a6ee.png)

**Comment by Hahahayabanan** [comment]
Модификаторы в отрыве от самого элемента

![image](https://user-images.githubusercontent.com/43137324/90308928-e692fa80-def4-11ea-98c4-d15356f0538f.png)
